translators/DirectionalLight.cpp

```diff
@@ -0,0 +1,310 @@
+MStatus DirectionalLight::updateMayaAttributes(MObject mayaObj, const UsdPrim& prim)
+{
+    MStatus status = MS::kSuccess;
+
+    const char* const errorString = "DirectionalLightTranslator: error setting maya directional light parameters";
+
+    UsdLuxDistantLight usdDistLight(prim);
+
+    GfVec3f pointWorld(1.0, 1.0, 1.0);
+    float angle = 0.0;
+    float intensity = 0.0;
+    float exposure = 0.0;
+    float diffuse = 0.0;
+    float specular = 0.0;
+    bool normalize = false;
+    GfVec3f color(1.0, 1.0, 1.0);
+    bool enableColorTemperature = false;
+
+    if(usdDistLight.GetPrim().HasAttribute(TfToken("pointWorld")))
+        usdDistLight.GetPrim().GetAttribute(TfToken("pointWorld")).Get(&pointWorld);
+    usdDistLight.GetAngleAttr().Get(&angle);
+    usdDistLight.GetIntensityAttr().Get(&intensity); //right now just use Maya's intensity value, not using the lightIntensity
+    usdDistLight.GetExposureAttr().Get(&exposure);
+    usdDistLight.GetDiffuseAttr().Get(&diffuse);
+    usdDistLight.GetSpecularAttr().Get(&specular);
+    usdDistLight.GetNormalizeAttr().Get(&normalize);
+    usdDistLight.GetColorAttr().Get(&color);
+    usdDistLight.GetEnableColorTemperatureAttr().Get(&enableColorTemperature);
+
+     //Update prim attribute values to maya node attibute values
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setVec3(mayaObj,m_pointWorld,pointWorld[0], pointWorld[1], pointWorld[2]), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setAngle(mayaObj,m_lightAngle,MAngle(angle,MAngle::Unit::kRadians)), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setVec3(mayaObj,m_color, color[0], color[1], color[2]), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setFloat(mayaObj,m_intensity, intensity), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setFloat(mayaObj,m_exposure, exposure), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setFloat(mayaObj,m_diffuse, diffuse), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setFloat(mayaObj,m_specular, specular), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setBool(mayaObj,m_normalize, normalize), errorString);
+    AL_MAYA_CHECK_ERROR(DgNodeTranslator::setBool(mayaObj,m_enableColorTemperature, enableColorTemperature), errorString);
+
+    return status;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+bool DirectionalLight::updateUsdPrim(const UsdStageRefPtr& stage, const SdfPath& usdPath, const MObject& mayaObj)
+{
+    const char* const errorString = "DirectionalLightTranslator: error getting maya directional light parameters";
+    float pointWorld[3] = {1.0,1.0,1.0};
+    MAngle angle;
+    float intensity = 0.0;
+    float exposure = 0.0;
+    float diffuse = 1.0;
+    float specular = 1.0;
+    bool normalize = false;
+    float color[3] = {1.0,1.0,1.0};
```

Comment:
Always postfix floating point constants with 'f'

```suggestion
    float color[3] = {1.0f, 1.0f, 1.0f};
```

Reply:
Yes, I will change it.